docs/moosedocs.py

```diff
@@ -2,10 +2,15 @@
 import sys
 import os
 import pdb
+import re
 
 # Locate MOOSE directory
 MOOSE_DIR = os.getenv('MOOSE_DIR', os.path.join(os.getcwd(), '..', 'moose'))
 if not os.path.exists(MOOSE_DIR):
+  RE = re.compile(r'(.*?/moose)(.*)')
+  matches = RE.search(os.getcwd())
+  MOOSE_DIR = matches.group(1)
+if not os.path.exists(MOOSE_DIR):
```

Comment:
What's going on here is that the first attempt to make MOOSE_DIR does so by making a path "/Users/tolmmm/Projects/moose/docs/../moose" because the current directory when this is called is "/Users/tolmmm/Projects/moose/docs".  So since the moose directory isn't in the moose directory, that fails.

Then the backup check doesn't work for me because my directory structure has "Projects" capitalized.  I have noticed it until now, because somehow everything still works with MOOSE_DIR being set as "/Users/tolmmm/projects/moose".  But the one thing that doesn't work is the moosedocs.py test, because then the gold file has "Projects" when it grabs the directory structure, but MOOSE_DIR has "projects".

So I added a thing here that works similarly to the gold file.  I felt it was the most robust way to go about things in case someone else comes along who also doesn't set up the directory to be exactly "/Users/<user_name>/projects/moose".